tests/unit/s2n_ktls_feature_probe_test.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "tls/s2n_ktls.h"
+
+#if defined(__linux__)
+    #include "linux/version.h"
+#endif
+
+int main(int argc, char **argv)
+{
```

Comment:
This works for now, but I'm not sure it needs to be its own test file long term. I'd probably just include it in one of your other ktls test files.

Reply:
A trick to checking that kTLS supported vs not supported is to `undef S2N_PLATFORM_SUPPORTS_KTLS` at the top of files and check the other case.

I am going to need to do that for other kTLS unit tests also and would like to avoid fiddling with the feature probe test each time. So we can move this to the same file later but I would prefer to keep it in a separate file during development.

Reply:
I would not suggest using that method, even for manual testing. You're going to get unexpected behavior, since the flag will be set some places but not others, and the visibility is pretty bad. A more robust solution would be to make s2n_ktls_is_supported_on_platform reference a global variable and add a method to set that global variable when testing.

Reply:
Global var seem like an anti pattern. Additionally we would then need to check both the `global var + S2N_PLATFORM_SUPPORTS_KTLS`. Testing code is now mixing with real code and that seems bad.

Would we only use global var for the testing check? Where do global variables live, or simply in s2n_ktls?

Reply:
Spoke offline. I will be using `undef` only for manual local testing since its not possible to compile ktls code on unsupported platforms. I added a task to combine the tests into a single file post development.